### DISCUSSION TITLE
Problem with connection using username and pass

### DISCUSSION THREAD

**Discussion by ofvicente** [Hugovdberg/PIconnect#640] [Q&A]
Hi,

I'm experiencing problems while including username and password calling PiConnect.

As an example:

`with PIconnect.PI.PIServer(server=None, 
username="test", password="test", 
 authentication_mode=AuthenticationMode.PI_USER_AUTHENTICATION, timeout=None) as server:
 print(server.server_name) `

The error returned is: 

Connect() takes 2 positional arguments but 3 were given

![image](https://user-images.githubusercontent.com/60259840/174275219-85465cac-4f89-49bc-baf6-d991cdf54450.png)

If I remove the user and pass args, it seems to work properly. I've tried from a new conda env, and also including a server name but the problem persists. 

Thanks in advance.

**Comment by Hugovdberg** [comment]
That is a strange issue, because `self._credentials` is either `None` or a tuple of a `NetworkCredential` and an integer. I have no idea what third argument is passed, unless it is the `self` argument, but I think that should be handled by the `pythonnet` library used to load the SDK into Python. The problem with this method is that I cannot test it with username and password on my system, so it is actually all code tested manually by other users :-(

**Comment by dpcamargo** [comment]
I had the same issue and it fixed after I installed the PI-AF-Client and PISDK. I noticed one of them was missing because I wasnt getting the PIConnect message when importing it's module "OSIsoft(r) AF SDK Version: 2.10.9.593" on python.

**Comment by Hugovdberg** [comment]
Ah, another case of the weird bugs caused by my heuristic "if I can't find the SDK I must be in testing mode"... PIconnect 0.10, which I just released, should prevent a lot of these issues by using more specific heuristics, and raising an error if the SDK is not found instead of silently failing.
@ofvicente could you check again with the latest version of PIconnect?
